source/grid/grid_generator.cc

```diff
@@ -1910,6 +1910,259 @@ namespace GridGenerator
+               ++face_n)
+            {
+              const auto face = cell->face(face_n);
+              if (face->at_boundary())
+                {
+                  const Point<2> center = face->center();
+                  // left side
+                  if (std::abs(center[0] - bl[0]) < 1e-10)
+                    face->set_boundary_id(0);
+                  // right side
+                  else if (std::abs(center[0] - tr[0]) < 1e-10)
+                    face->set_boundary_id(1);
+                  // bottom
+                  else if (std::abs(center[1] - bl[1]) < 1e-10)
+                    face->set_boundary_id(2);
+                  // top
+                  else if (std::abs(center[1] - tr[1]) < 1e-10)
+                    face->set_boundary_id(3);
+                  // cylinder boundary
+                  else
+                    {
+                      Assert(cell->material_id() == 2, ExcInternalError());
+                      face->set_boundary_id(4);
+                    }
+                }
+            }
+
+      // move to the new center
+      GridTools::shift(new_center, tria);
+
+      PolarManifold<2> polar_manifold(new_center);
+      tria.set_manifold(polar_manifold_id, polar_manifold);
+      TransfiniteInterpolationManifold<2> inner_manifold;
+      inner_manifold.initialize(tria);
+      tria.set_manifold(tfi_manifold_id, inner_manifold);
+    }
+
+  } // namespace internal
+
+  template <>
+  void plate_with_a_hole(Triangulation<2> &       tria,
+                         const double             inner_radius,
+                         const double             outer_radius,
+                         const double             pad_bottom,
+                         const double             pad_top,
+                         const double             pad_left,
+                         const double             pad_right,
+                         const Point<2>           new_center,
+                         const types::manifold_id polar_manifold_id,
+                         const types::manifold_id tfi_manifold_id,
+                         const double             L,
+                         const unsigned int       n_slices,
+                         const bool               colorize)
+  {
+    internal::plate_with_a_hole(tria,
+                                inner_radius,
+                                outer_radius,
+                                pad_bottom,
+                                pad_top,
+                                pad_left,
+                                pad_right,
+                                new_center,
+                                polar_manifold_id,
+                                tfi_manifold_id,
+                                L,
+                                n_slices,
+                                colorize);
+
+    // reset material id
+    for (auto &cell : tria.active_cell_iterators())
+      cell->set_material_id(0);
```

Comment:
How come this is not a part of the internal function?

Reply:
because I need material id's there for 3D case.